lab_2/filtration.py

```diff
@@ -0,0 +1,208 @@
+    result_array = []
+    psf = np.ones((3, 3)) / 25
+    for i in tqdm(range(0, len(files)), desc="Фильтрация: "):
+        img = img_as_float(rgb2gray(files[i]))
+        restored_img = restoration.richardson_lucy(img, psf)
+        restored_img = img_as_ubyte(
+            (restored_img - np.min(restored_img)) / (np.max(restored_img) - np.min(restored_img)))
+        result_array.append(restored_img)
+        progress_bar['value'] += progress_step
+        progress_label.config(text=round(progress_bar['value']))
+        root.update_idletasks()
+        # io.imsave("temp/" + str(i) + ".jpg", restored_img)
+    progress_bar['value'] = 100
+    progress_label.config(text=progress_bar['value'])
+    root.update_idletasks()
+    return result_array
+
+
+def wavelet(files, progress_bar, progress_label, root):
+    from skimage.restoration import denoise_wavelet
+    progress_step = 100 / len(files)
+    progress_bar['value'] = 0
+    progress_label.config(text="0")
+    root.update_idletasks()
+    result_array = []
+    for i in tqdm(range(0, len(files)), desc="Фильтрация: "):
+        img = img_as_float(rgb2gray(files[i]))
+        restored_img = denoise_wavelet(img, rescale_sigma=True)
+        restored_img = img_as_ubyte(
+            (restored_img - np.min(restored_img)) / (np.max(restored_img) - np.min(restored_img)))
+        result_array.append(restored_img)
+        progress_bar['value'] += progress_step
+        progress_label.config(text=round(progress_bar['value']))
+        root.update_idletasks()
+        # io.imsave("temp/" + str(i) + ".jpg", restored_img)
+    progress_bar['value'] = 100
+    progress_label.config(text=progress_bar['value'])
+    root.update_idletasks()
+    return result_array
+
+
+def filtration_gui_main(files, mode, progress_bar_info):
+    result_array = []
+    if mode == "Без предобработки":
+        result_array = files
+    if mode == "Винер":
+        result_array = wiener(files, progress_bar_info[0], progress_bar_info[1], progress_bar_info[2])
+    if mode == "Гаусс":
+        result_array = gauss(files, progress_bar_info[0], progress_bar_info[1], progress_bar_info[2])
+    if mode == "Медианный":
+        result_array = median(files, progress_bar_info[0], progress_bar_info[1], progress_bar_info[2])
+    if mode == "Контраст":
+        result_array = contrast(files, progress_bar_info[0], progress_bar_info[1], progress_bar_info[2])
+    if mode == "Резкость":
+        result_array = sharpen(files, progress_bar_info[0], progress_bar_info[1], progress_bar_info[2])
+    if mode == "Шумоподавление":
+        result_array = denoise(files, progress_bar_info[0], progress_bar_info[1], progress_bar_info[2])
+    if mode == "Обратная свёртка":
+        result_array = deconvolution(files, progress_bar_info[0], progress_bar_info[1], progress_bar_info[2])
+    if mode == "Вейвлет":
+        result_array = wavelet(files, progress_bar_info[0], progress_bar_info[1], progress_bar_info[2])
+    return result_array
```

Comment:
Функцию main как-то можно отрефакторить?

Reply:
Да, проглядел. Явное нарушение принципа открытости/закрытости. Можно использовать словарь, дабы вынести все режимы работы в одно место, где, при необходимости, можно будет добавлять новые режимы, без изменения логики. 

```suggestion
def filtration_gui_main(files, mode, progress_bar_info):

    modes = {
        "Винер": wiener,
        "Гаусс": gauss,
        "Медианный": median,
        "Контраст": contrast,
        "Резкость": sharpen,
        "Шумоподавление": denoise,
        "Обратная свёртка": deconvolution,
        "Вейвлет": wavelet
    }

    func = modes.get(mode)
    
    if func is None:
        raise ValueError("Несуществующий режим")
        
    if mode == "Без предобработки":
        return func(files)
        
    else:
        return func(files, progress_bar_info[0], progress_bar_info[1], progress_bar_info[2])
```